cmd/run/run_test.go

```diff
@@ -1167,3 +1173,102 @@ func TestRunCommandConfigIsMerged(t *testing.T) {
 	rootCmd.SetArgs([]string{"run"})
 	require.NoError(t, rootCmd.Execute())
 }
+
+func TestHTTPHeaders(t *testing.T) {
+	cfg := MustDefaultConfigWithRandomPorts()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
```

Comment:
**(non-blocking, suggestion):** I'm noticing we're using defer a lot in tests across the code base. Have we considered switching to `t.Cleanup` instead?

The advantage that it brings is that it's guaranteed to run after the test or benchmark is finished, even if `t.FailNow()`,` t.Fatal()`, `t.SkipNow()`, or `t.Skip() `is called, which is not the case with `defer`.

Reply:
🤔  it is called

<img width="1913" alt="image" src="https://github.com/openfga/openfga/assets/5374887/238574bc-bc0d-4745-9677-93ba64f1ba98">

Reply:
Oh that's interesting! Thanks for checking 👍🏻 . I'll probably need to get updated on this. I'll research it a bit more.

Reply:
@miparnisari alright did a bit of digging to brush off some of the wrong assumptions I had on this.

The main benefit is around execution timing: `defer` executes when the function it is called in returns. `t.Cleanup` executes when the test and all its subtests have completed.

So we can think of `t.Cleanup()` as an enhanced and extended version of `defer`. It also documents that the passed functions are for cleanup purposes.

Here's also a good article on the topic: https://brandur.org/fragments/go-prefer-t-cleanup-with-parallel-subtests.